src/components/PDFView/index.native.js

```diff
@@ -1,50 +1,176 @@
+        this.setState({
+            password,
+            shouldAttemptPdfLoad: true,
+            shouldShowLoadingIndicator: true,
+            isPasswordInvalid: false,
+        });
+    }
+
+    /**
+     * After the PDF is successfully loaded hide PDFPasswordForm and the loading
+     * indicator.
+     */
+    finishPdfLoad() {
+        this.setState({
+            shouldRequestPassword: false,
+            shouldShowLoadingIndicator: false,
+        });
+    }
+
+    render() {
+        const pdfStyles = [
+            styles.imageModalPDF,
+            StyleUtils.getWidthAndHeightStyle(this.props.windowWidth, this.props.windowHeight),
+        ];
+        const touchableStyles = [
+            styles.flex1,
+            this.props.style,
+        ];
+
+        // If we haven't yet successfully validated the password and loaded the PDF,
+        // then we need to hide the react-native-pdf/PDF component so that PDFPasswordForm
+        // is positioned nicely. We're specifically hiding it because we still need to render
+        // the PDF so that it can validate the password.
+        if (this.state.shouldRequestPassword) {
+            pdfStyles.push(styles.invisible);
+        }
+
+        // For small screens force the container view to take up the full width when
+        // displaying the password form.
+        const containerStyles = this.state.shouldRequestPassword && this.props.isSmallScreenWidth
+            ? styles.pdfPasswordForm.nativeNarrowContainer : {};
+
+        const keyboardBehavior = CONST.PDF_PASSWORD_FORM.KEYBOARD_AVOIDING_VIEW.BEHAVIOR;
+        const keyboardOffset = CONST.PDF_PASSWORD_FORM.KEYBOARD_AVOIDING_VIEW.KEYBOARD_VERTICAL_OFFSET;
 
-const PDFView = props => (
-    <TouchableWithoutFeedback style={[styles.flex1, props.style]}>
-        <PDF
-            activityIndicator={<FullScreenLoadingIndicator />}
-            source={{uri: props.sourceURL}}
-            style={[
-                styles.imageModalPDF,
-                StyleUtils.getWidthAndHeightStyle(props.windowWidth, props.windowHeight),
-            ]}
-        />
-    </TouchableWithoutFeedback>
-);
+        return (
+            <View style={containerStyles}>
+                {this.state.shouldAttemptPdfLoad && (
+                    <TouchableWithoutFeedback style={touchableStyles}>
+                        <PDF
+                            activityIndicator={<FullScreenLoadingIndicator />}
+                            source={{uri: this.props.sourceURL}}
+                            style={pdfStyles}
+                            onError={this.initiatePasswordChallenge}
+                            password={this.state.password}
+                            onLoadComplete={this.finishPdfLoad}
+                        />
+                    </TouchableWithoutFeedback>
+                )}
+                {this.state.shouldRequestPassword && (
+                    <KeyboardAvoidingView
```

Comment:
Use our custom component instead `KeyboardAvoidingView` if that works here.

Reply:
I can't use `components/KeyboardAvoidingView` because all it's properties are hard coded.

Should I update `components/KeyboardAvoidingView` so that it's more flexible?

Reply:
> I can't use components/KeyboardAvoidingView because all its properties are hard coded.

Why do you have to customize it? Can you explain me what would be effect of each change?

Reply:
We need `keyboardVerticalOffset`:
```
<KeyboardAvoidingView
    ...
    keyboardVerticalOffset={keyboardOffset}
```
Without `keyboardVerticalOffset` the confirm button is overlapped by the keyboard.

Reply:
Hmm, I see. Which Platform?

Reply:
Yeah, it's for iOS. Without this the confirm button isn't positioned correctly when the keyboard is open. Please note that this is only for native - this has no effect on mweb.